clang/lib/Analysis/CallGraph.cpp

```diff
@@ -136,6 +137,19 @@ class CGBuilder : public StmtVisitor<CGBuilder> {
     }
   }
 
+  void VisitIfStmt(IfStmt *If) {
+    if (G->shouldSkipConstantExpressions()) {
+      if (llvm::Optional<Stmt *> ActiveStmt =
+              If->getNondiscardedCase(G->getASTContext())) {
+        if (*ActiveStmt)
+          this->Visit(*ActiveStmt);
+        return;
+      }
+    }
+
+    VisitChildren(If);
```

Comment:
This line PERHAPS (i haven't thought enough through?) should be the `StmtVisitor::VisitStmt(If);` call instead... I think it ends up changing nothing, but is probably the right thing to do.